e2_tts_pytorch/dataset/e2_dataset.py

```diff
@@ -0,0 +1,70 @@
+import torch
+from torch.utils.data import Dataset
+import torchaudio
+from e2_tts_pytorch.utils.compute_mel import TorchMelSpectrogram
+from datasets import load_dataset
+import logging
+from einops import rearrange, reduce
+
+logger = logging.getLogger(__name__)
+
+class E2EDataset(Dataset):
+    def __init__(self, hf_dataset):
+        self.data = load_dataset(hf_dataset, split='train')
+        self.target_sample_rate = 22050
+        self.hop_length = 256
+        self.mel_spectrogram = TorchMelSpectrogram(sampling_rate=self.target_sample_rate)
+        
+        self.char_set = set("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789.,!?-:;'\"()[] ")
```

Comment:
do you want to try just using https://github.com/lucidrains/e2-tts-pytorch/blob/main/e2_tts_pytorch/e2_tts.py#L55 ?